### PR TITLE
Added restrictions and clarifications

### LINKED ISSUE #81: Spec: if containment can't apply then s-v has no effect (e.g. display: contents)
I don't think we can reasonably expect to lock a display: contents element, since it has no box for which children can be skipped. That being said, in the imperative version we've relied on promise rejection or displayLock.locked value to figure out whether we're actually locked.

What should we do in the declarative version? Should we just clear the rendersubtree attribute? Or do nothing?

/cc @chrishtr @rakina

### PR DESCRIPTION
**Pull request by vmpstr**
https://vmpstr.github.io/display-locking/index.html#restrictions

Closes #81 

Please take a look.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -195,6 +195,19 @@ the user-agent should retain the previously computed layout state if possible.
   values are added on top of any existing containment values.
 </div>
 
+Restrictions and Clarifications {#restrictions}
+===============================
+
+* In situations where [=layout containment=] has no effect (e.g. the
+    element does not generate a principal box), 'subtree-visibility' values also
+    have no effect. In other words, if [=layout containment=] has no effect, the
```
Comment (chrishtr) [inline index.bs]: Consider removing the second sentence, seems redundant.

File: index.bs [hunk 2]
```diff
@@ -195,6 +195,19 @@ the user-agent should retain the previously computed layout state if possible.
...
+    'subtree-visibility' value is effectively [=visible=].
+
+* <a href="https://html.spec.whatwg.org/multipage/rendering.html#replaced-elements">
```
Comment (chrishtr) [inline index.bs]: What about:

Object, embed, iframe?

File: index.bs (hunk 1 above)
Comment (vmpstr) [reply index.bs]: Done.

File: index.bs (hunk 2 above)
Comment (vmpstr) [reply index.bs]: We also don't paint iframe contents (we have tests for that). I'll add tests for object and embed, but I think we should not separate things into different subgroups of replaced if we don't have a reason to do so. So, I think this phrasing _should_ be correct. WDYT?

Comment (chrishtr) [reply index.bs]: Ok, but then you should either omit "such as images" to avoid it sounding like it's just for images.

File: index.bs [hunk 3]
```diff
@@ -195,6 +195,18 @@ the user-agent should retain the previously computed layout state if possible.
...
 
+Restrictions and Clarifications {#restrictions}
+===============================
+
+* In situations where [=layout containment=] has no effect (e.g. the
+    element does not generate a principal box), 'subtree-visibility' values also
+    have no effect.
+
+* <a href="https://html.spec.whatwg.org/multipage/rendering.html#replaced-elements">
+    Replaced elements</a>, such as images, do not paint their contents if they
```
Comment (chrishtr) [inline index.bs]: Suggest further specifying that "do not paint their contents" means not running the parts of step 7.1 or step 7.4 from this algorithm that refer to replaced contents:

https://www.w3.org/TR/CSS21/zindex.html

Comment (vmpstr) [reply index.bs]: Done. Let me know if this reads ok.

**Comment by vmpstr** [comment]
ptal

**Pull request merged**